server/app/views/dev/ThymeleafComponentsDemo.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="http://www.thymeleaf.org">
+<head>
+  <meta charset="UTF-8">
+  <title>Title</title>
+</head>
+<body>
+
+<button th:fragment="toggle-on"
+        class="flex gap-2 items-center text-gray-400 font-medium bg-transparent rounded-full hover:bg-gray-400 hover:text-gray-300"
+        hx-swap="outerHTML"
+        hx-get="/dev/thymeleafDemo/components/toggle/off">
+  <p class="hover-group:text-white">optional</p>
+  <input hidden="" name="toggle" value="on">
```

Comment:
To clarify my understanding of how this works:
 - This toggle starts as on.
 - When it's clicked, the `hx-get` attribute is used to re-render this part of the HTML. (side question: Should this be a POST since it's modifying application state?)
 - That `hx-get` attribute is `"/dev/thymeleafDemo/components/toggle/off"`, which corresponds to `ThymeleafDemoController#componentToggle(Http.Request request, String state)` called with `state="off"`.
 - The controller re-processes this `ThymeleafComponentsDemo` template with a `"toggle-off"` parameter...

And this is where I lost of the flow of it :) How does thymeleaf (or HTMX?) know to replace just the `th:fragment="toggle-on"` part of the HTML? How does the `<input>` toggle element change its value to be `value="off"`?

Reply:
The latter is my understanding:
- The initial template renders with both the `toggle-on` and `toggle-off` components, insterted using the `th:insert` attribute
- When the button is pressed, it triggers htmx to make a get request to and endpoint that only renders the template for that `toggle-on` (or `-off`) component. htmx then swaps out the contents of that portion of the DOM with the response from the `/dev/thymeleafDemo/components/toggle/:state` endpoint